1.6.17 release doesn't enable on Confluence

Throws the following:

```
2022-08-18 13:43:40,913 ERROR [ThreadPoolAsyncTaskExecutor::Thread 38] [internal.dependencies.startup.DependencyWaiterApplicationContextExecutor] fail Unable to create application context for [com.atlassian.plugins.confluence.markdown.confluence-markdown-macro], unsatisfied dependencies: none
 -- url: /confluence/rest/plugins/1.0/ | referer: https://lab.remote-desktop.org/confluence/plugins/servlet/upm/marketplace/search?q=markdown | traceId: 3e89b7cc725fa7fb | userName: admin
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'migrationListener': Lookup method resolution failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [com.atlassian.plugins.confluence.markdown.ccma.MigrationListener] from ClassLoader [com.atlassian.plugins.confluence.markdown.confluence-markdown-macro [315]]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:289)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineConstructorsFromBeanPostProcessors(AbstractAutowireCapableBeanFactory.java:1302)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
	at org.eclipse.gemini.blueprint.context.support.AbstractDelegatedExecutionApplicationContext.access$1600(AbstractDelegatedExecutionApplicationContext.java:57)
	at org.eclipse.gemini.blueprint.context.support.AbstractDelegatedExecutionApplicationContext$4.run(AbstractDelegatedExecutionApplicationContext.java:322)
	at org.eclipse.gemini.blueprint.util.internal.PrivilegedUtils.executeWithCustomTCCL(PrivilegedUtils.java:85)
	at org.eclipse.gemini.blueprint.context.support.AbstractDelegatedExecutionApplicationContext.completeRefresh(AbstractDelegatedExecutionApplicationContext.java:287)
	at org.eclipse.gemini.blueprint.extender.internal.dependencies.startup.DependencyWaiterApplicationContextExecutor$CompleteRefreshTask.run(DependencyWaiterApplicationContextExecutor.java:137)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.atlassian.plugins.confluence.markdown.ccma.MigrationListener] from ClassLoader [com.atlassian.plugins.confluence.markdown.confluence-markdown-macro [315]]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:485)
	at org.springframework.util.ReflectionUtils.doWithLocalMethods(ReflectionUtils.java:321)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:267)
	... 18 more
Caused by: java.lang.LinkageError: loader constraint violation: loader org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @38ad5f65 wants to load interface com.atlassian.migration.app.gateway.AppCloudMigrationGateway. A different interface with the same name was previously loaded by org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @134b02c4. (com.atlassian.migration.app.gateway.AppCloudMigrationGateway is in unnamed module of loader org.apache.felix.framework.BundleWiringImpl$BundleClassLoader @134b02c4, parent loader org.apache.catalina.loader.ParallelWebappClassLoader @609ed3d4)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.defineClass(BundleWiringImpl.java:2410)
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.findClass(BundleWiringImpl.java:2194)
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1607)
	at org.apache.felix.framework.BundleWiringImpl.access$200(BundleWiringImpl.java:80)
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2053)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:467)
	... 20 more
```

Reproduced on lab, and caused an issue at:
https://atlasauthority.atlassian.net/browse/SUPPORT-1770
https://atlasauthority.atlassian.net/browse/SUPPORT-1774
https://atlasauthority.atlassian.net/browse/SUPPORT-1781

We have pulled the 1.6.17 release from marketplace entirely as a result.

--- a/pom.xml
+++ b/pom.xml
@@ -190,13 +190,6 @@
                     <!-- https://developer.atlassian.com/docs/advanced-topics/configuration-of-instructions-in-atlassian-plugins -->
                     <instructions>
                         <Atlassian-Plugin-Key>${atlassian.plugin.key}</Atlassian-Plugin-Key>
-
-                        <Export-Package>
-                            com.atlassian.migration.app.listener,
-                            com.atlassian.migration.app.gateway,
-                            com.atlassian.migration.app,
-                        </Export-Package>
-
                         <!-- Add package import here -->
                         <Import-Package>
                             com.atlassian.*;resolution:="optional",

--- a/src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java
+++ b/src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java
@@ -100,7 +100,7 @@ public class MarkdownFromURLMacro extends BaseMacro implements Macro {
 	}
 
     @Autowired
-    public MarkdownFromURLMacro(@ComponentImport PageBuilderService pageBuilderService, XhtmlContent xhtmlUtils, BandanaManager bandanaManager) {
+    public MarkdownFromURLMacro(@ComponentImport PageBuilderService pageBuilderService, @Autowired XhtmlContent xhtmlUtils, @Autowired BandanaManager bandanaManager) {
         this.pageBuilderService = pageBuilderService;
         this.xhtmlUtils = xhtmlUtils;
         this.bandanaManager = bandanaManager;

--- a/src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownMacro.java
+++ b/src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownMacro.java
@@ -54,7 +54,7 @@ public class MarkdownMacro extends BaseMacro implements Macro {
     private ConfluenceBandanaContext context = new ConfluenceBandanaContext("markdown-plugin");
 
     @Autowired
-    public MarkdownMacro(@ComponentImport PageBuilderService pageBuilderService, XhtmlContent xhtmlUtils, BandanaManager bandanaManager) {  
+    public MarkdownMacro(@ComponentImport PageBuilderService pageBuilderService, @ComponentImport XhtmlContent xhtmlUtils, @ComponentImport BandanaManager bandanaManager) {
         this.pageBuilderService = pageBuilderService;
         this.xhtmlUtils = xhtmlUtils;
         this.bandanaManager = bandanaManager;

